hadoop-plugin/src/main/groovy/com/linkedin/gradle/hdfs/HdfsFileSystem.groovy

```diff
@@ -192,6 +196,47 @@ class HdfsFileSystem {
   }
 
   /**
+   * The zipFile is the zip file on the local system. This file is extracted to the destDirectory.
+   * If destDirectory doesn't exist, then it is created.
+   *
+   * @param zipFile The zip file on local system
+   * @param destDirectory The destination directory on hdfs where zipFile should be extracted.
+   * @throws IOException If zipFile cannot be extracted to destDirectory
+   */
+  public void copyFromLocalZip(File zipFile, Path destDirectory) throws IOException {
+
+    // create the destination directory if it doesn't exists
+    if (!exists(destDirectory)) {
+      mkdir(destDirectory);
+    }
+
+    logger.info("zip file path: ${zipFile.toString()}");
+    logger.info("destination directory: ${destDirectory.toString()}");
+
+    ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFile));
+    ZipEntry entry = zipIn.getNextEntry();
+
+    //iterates over entries in the zip file
+    while (entry != null) {
+      String filePath = destDirectory.toString() + File.separator + entry.getName();
```

Comment:
Does entry.getName() include the path information in the zip? Let's say that in the zip the entry has the path /src/main/pig/myScript.pig. On HDFS, does it get copied to ${destDirectory}/src/main/pig/myScript.pig or just to ${destDirectory}/myScript.pig? I hope this preserves the path information (we must find a way to preserve it).

Super glad you have found a way to do this without extracting all the files from the zip into a temp directory - I really hope this approach allows us to preserve the path information in the zip as we copy it to the dest directory on HDFS!